### PR TITLE
Re-add server up metrics

### PR DESCRIPTION
**Pull request by coder-hugo**
### What does this PR do?

This PR re-adds the server up metric to traefik 2.x

### Motivation

The server up metric is necessary for our setup

### More

- [x] Added/updated tests
- [ ] Added/updated documentation

### CODE REVIEW

File: pkg/server/router/router.go [hunk 1]
```diff
@@ -41,6 +42,7 @@ type Manager struct {
 	chainBuilder       *middleware.ChainBuilder
 	modifierBuilder    responseModifierBuilder
 	conf               *runtime.Configuration
+	metricsRegistry    metrics.Registry
```
Comment (juliens) [inline pkg/server/router/router.go:45]: As explained in the previous comment, could you remove it?

File: pkg/server/router/router.go [hunk 2]
```diff
@@ -49,6 +51,7 @@ func NewManager(conf *runtime.Configuration,
 	middlewaresBuilder middlewareBuilder,
 	modifierBuilder responseModifierBuilder,
 	chainBuilder *middleware.ChainBuilder,
+	metricsRegistry metrics.Registry,
```
Comment (juliens) [inline pkg/server/router/router.go:54]: As explained in the previous comment, could you remove it?

File: pkg/server/router/router.go [hunk 3]
```diff
@@ -57,6 +60,7 @@ func NewManager(conf *runtime.Configuration,
 		modifierBuilder:    modifierBuilder,
 		chainBuilder:       chainBuilder,
 		conf:               conf,
+		metricsRegistry:    metricsRegistry,
```
Comment (juliens) [inline pkg/server/router/router.go:63]: As explained in the previous comment, could you remove it?

File: pkg/server/routerfactory.go [hunk 4]
```diff
@@ -25,14 +26,15 @@ type RouterFactory struct {
 	entryPointsTCP []string
 	entryPointsUDP []string
 
-	managerFactory *service.ManagerFactory
+	managerFactory  *service.ManagerFactory
+	metricsRegistry metrics.Registry
```
Comment (juliens) [inline pkg/server/routerfactory.go:30]: As explained in the previous comment, could you remove it?

File: pkg/server/routerfactory.go [hunk 5]
```diff
@@ -25,14 +26,15 @@ type RouterFactory struct {
...
 
 	chainBuilder *middleware.ChainBuilder
 	tlsManager   *tls.Manager
 }
 
 // NewRouterFactory creates a new RouterFactory
-func NewRouterFactory(staticConfiguration static.Configuration, managerFactory *service.ManagerFactory, tlsManager *tls.Manager, chainBuilder *middleware.ChainBuilder) *RouterFactory {
+func NewRouterFactory(staticConfiguration static.Configuration, managerFactory *service.ManagerFactory, tlsManager *tls.Manager, chainBuilder *middleware.ChainBuilder, metricsRegistry metrics.Registry) *RouterFactory {
```
Comment (juliens) [inline pkg/server/routerfactory.go:37]: As explained in the previous comment, could you remove the metrics registry?

File: pkg/server/routerfactory.go [hunk 6]
```diff
@@ -49,11 +51,12 @@ func NewRouterFactory(staticConfiguration static.Configuration, managerFactory *
...
 	return &RouterFactory{
-		entryPointsTCP: entryPointsTCP,
-		entryPointsUDP: entryPointsUDP,
-		managerFactory: managerFactory,
-		tlsManager:     tlsManager,
-		chainBuilder:   chainBuilder,
+		entryPointsTCP:  entryPointsTCP,
+		entryPointsUDP:  entryPointsUDP,
+		managerFactory:  managerFactory,
+		metricsRegistry: metricsRegistry,
```
Comment (juliens) [inline pkg/server/routerfactory.go:57]: As explained in the previous comment, could you remove it?

File: pkg/server/routerfactory.go [hunk 7]
```diff
@@ -69,7 +72,7 @@ func (f *RouterFactory) CreateRouters(conf dynamic.Configuration) (map[string]*t
 	middlewaresBuilder := middleware.NewBuilder(rtConf.Middlewares, serviceManager)
 	responseModifierFactory := responsemodifiers.NewBuilder(rtConf.Middlewares)
 
-	routerManager := router.NewManager(rtConf, serviceManager, middlewaresBuilder, responseModifierFactory, f.chainBuilder)
+	routerManager := router.NewManager(rtConf, serviceManager, middlewaresBuilder, responseModifierFactory, f.chainBuilder, f.metricsRegistry)
```
Comment (juliens) [inline pkg/server/routerfactory.go:75]: As explained in the previous comment, could you remove metrics registry?

**Review by rtribotte — APPROVED** [review]
LGTM 👍

**Review by jbdoumenjou — APPROVED** [review]
LGTM

**Review by juliens — APPROVED** [review]
LGTM

**Comment by ldez** [comment]
SemaphoreCI has a problem related to [the change of the organization](https://github.com/traefik/traefik/issues/7293), I will have to close and then reopen your PR.
Sorry for the inconvenience

**Comment by rcomanne** [comment]
So, this should be included in the 2.4.0-rc1 release correct?
When testing locally with a docker-compose, I do not see the metrics being exposed by Prometheus.
Is the documentation missing a new entry on how to enable these metrics?

**Pull request merged**